### PR TITLE
Regenerate gearman doc to fix #2859

### LINKED ISSUE #2859: GearmanTask::jobHandle() can also return `false`
From manual page: https://php.net/gearmantask.jobhandle

---

As you can see [in the source code](https://github.com/php/pecl-networking-gearman/blob/master/php_gearman_task.c#L205), this method normally returns a string, but can return `false` on failure. So the return type should be `string|false` but is just `string` on the manual page.

Edit: I just saw, it could also return `null`.

### PR DESCRIPTION
**Pull request by cracksalad**
Fixes #2859 
See also [php/pecl-networking-gearman PR#24](https://github.com/php/pecl-networking-gearman/pull/24)

Regenerated gearman doc based on PR#24 in pecl-networking-gearman to fix some return types in PHP manual pages. I ran exactly this command with the PR cloned:

`../php-src/build/gen_stub.php --replace-methodsynopses ../path/to/pecl-networking-gearman/ reference/gearman/`

@Girgias

### CODE REVIEW

File: reference/gearman/gearmanclient/jobstatus.xml [hunk 1]
```diff
@@ -11,10 +10,10 @@
  <refsect1 role="description">
   &reftitle.description;
   <para>&style.oop; (method):</para>
-   <methodsynopsis>
+   <methodsynopsis role="GearmanClient">
    <modifier>public</modifier> <type>array</type><methodname>GearmanClient::jobStatus</methodname>
    <methodparam><type>string</type><parameter>job_handle</parameter></methodparam>
-   </methodsynopsis>
+  </methodsynopsis>
```
Comment (Girgias) [inline reference/gearman/gearmanclient/jobstatus.xml:16]: The indent here is slightly broken

File: reference/gearman/gearmanjob/functionname.xml [hunk 2]
```diff
@@ -9,9 +8,9 @@
 
  <refsect1 role="description">
   &reftitle.description;
-  <methodsynopsis>
-   <modifier>public</modifier> <type>string</type><methodname>GearmanJob::functionName</methodname>
-   <void />
+  <methodsynopsis role="GearmanJob">
+   <modifier>public</modifier> <type class="union"><type>false</type><type>string</type></type><methodname>GearmanJob::functionName</methodname>
```
Comment (Girgias) [inline reference/gearman/gearmanjob/functionname.xml:12]: Return value section needs to be updated for false return value

File: reference/gearman/gearmanclient/addtaskstatus.xml [hunk 3]
```diff
@@ -9,10 +8,10 @@
 
  <refsect1 role="description">
   &reftitle.description;
-  <methodsynopsis>
-   <modifier>public</modifier> <type>GearmanTask</type><methodname>GearmanClient::addTaskStatus</methodname>
+  <methodsynopsis role="GearmanClient">
+   <modifier>public</modifier> <type class="union"><type>GearmanTask</type><type>false</type></type><methodname>GearmanClient::addTaskStatus</methodname>
```
Comment (Girgias) [inline reference/gearman/gearmanclient/addtaskstatus.xml:12]: Return value section needs to be update to mention the false return

File: reference/gearman/gearmanclient/error.xml [hunk 4]
```diff
@@ -9,9 +8,9 @@
 
  <refsect1 role="description">
   &reftitle.description;
-  <methodsynopsis>
-   <modifier>public</modifier> <type>string</type><methodname>GearmanClient::error</methodname>
-   <void />
+  <methodsynopsis role="GearmanClient">
+   <modifier>public</modifier> <type class="union"><type>string</type><type>false</type></type><methodname>GearmanClient::error</methodname>
```
Comment (Girgias) [inline reference/gearman/gearmanclient/error.xml:12]: Return value section needs to be updated for false return value

File: reference/gearman/gearmanjob/handle.xml [hunk 5]
```diff
@@ -9,9 +8,9 @@
 
  <refsect1 role="description">
   &reftitle.description;
-  <methodsynopsis>
-   <modifier>public</modifier> <type>string</type><methodname>GearmanJob::handle</methodname>
-   <void />
+  <methodsynopsis role="GearmanJob">
+   <modifier>public</modifier> <type class="union"><type>false</type><type>string</type></type><methodname>GearmanJob::handle</methodname>
```
Comment (Girgias) [inline reference/gearman/gearmanjob/handle.xml:12]: Return value section needs to be updated for false return value

File: reference/gearman/gearmanjob/unique.xml [hunk 6]
```diff
@@ -9,9 +8,9 @@
 
  <refsect1 role="description">
   &reftitle.description;
-  <methodsynopsis>
-   <modifier>public</modifier> <type>string</type><methodname>GearmanJob::unique</methodname>
-   <void />
+  <methodsynopsis role="GearmanJob">
+   <modifier>public</modifier> <type class="union"><type>false</type><type>string</type></type><methodname>GearmanJob::unique</methodname>
```
Comment (Girgias) [inline reference/gearman/gearmanjob/unique.xml:12]: Return value section needs to be updated for false return value

File: reference/gearman/gearmantask/functionname.xml [hunk 7]
```diff
@@ -9,9 +8,9 @@
 
  <refsect1 role="description">
   &reftitle.description;
-  <methodsynopsis>
-   <modifier>public</modifier> <type>string</type><methodname>GearmanTask::functionName</methodname>
-   <void />
+  <methodsynopsis role="GearmanTask">
+   <modifier>public</modifier> <type class="union"><type>false</type><type>string</type></type><methodname>GearmanTask::functionName</methodname>
```
Comment (Girgias) [inline reference/gearman/gearmantask/functionname.xml:12]: Return value section needs to be updated for false return value

File: reference/gearman/gearmantask/jobhandle.xml [hunk 8]
```diff
@@ -10,10 +9,10 @@
 
  <refsect1 role="description">
   &reftitle.description;
-   <methodsynopsis>
-   <modifier>public</modifier> <type>string</type><methodname>GearmanTask::jobHandle</methodname>
-   <void />
-   </methodsynopsis>
+   <methodsynopsis role="GearmanTask">
+   <modifier>public</modifier> <type class="union"><type>false</type><type>string</type></type><methodname>GearmanTask::jobHandle</methodname>
```
Comment (Girgias) [inline reference/gearman/gearmantask/jobhandle.xml:13]: Return value section needs to be updated for false return value

File: reference/gearman/gearmanworker/addoptions.xml [hunk 9]
```diff
@@ -9,8 +8,8 @@
 
  <refsect1 role="description">
   &reftitle.description;
-  <methodsynopsis>
-   <modifier>public</modifier> <type>bool</type><methodname>GearmanWorker::addOptions</methodname>
+  <methodsynopsis role="GearmanWorker">
+   <modifier>public</modifier> <type>true</type><methodname>GearmanWorker::addOptions</methodname>
```
Comment (Girgias) [inline reference/gearman/gearmanworker/addoptions.xml:12]: The return value section _could_ use the always.return.true XML entity defined in language-snippet

File: reference/gearman/gearmanworker/error.xml [hunk 10]
```diff
@@ -9,9 +8,9 @@
 
  <refsect1 role="description">
   &reftitle.description;
-  <methodsynopsis>
-   <modifier>public</modifier> <type>string</type><methodname>GearmanWorker::error</methodname>
-   <void />
+  <methodsynopsis role="GearmanWorker">
+   <modifier>public</modifier> <type class="union"><type>string</type><type>false</type></type><methodname>GearmanWorker::error</methodname>
```
Comment (Girgias) [inline reference/gearman/gearmanworker/error.xml:12]: Return value section needs to be updated for false return value

File: reference/gearman/gearmanworker/removeoptions.xml [hunk 11]
```diff
@@ -9,8 +8,8 @@
 
  <refsect1 role="description">
   &reftitle.description;
-  <methodsynopsis>
-   <modifier>public</modifier> <type>bool</type><methodname>GearmanWorker::removeOptions</methodname>
+  <methodsynopsis role="GearmanWorker">
+   <modifier>public</modifier> <type>true</type><methodname>GearmanWorker::removeOptions</methodname>
```
Comment (Girgias) [inline reference/gearman/gearmanworker/removeoptions.xml:12]: Same, could use the always true XML entity

File: reference/gearman/gearmanworker/setoptions.xml [hunk 12]
```diff
@@ -9,8 +8,8 @@
 
  <refsect1 role="description">
   &reftitle.description;
-  <methodsynopsis>
-   <modifier>public</modifier> <type>bool</type><methodname>GearmanWorker::setOptions</methodname>
+  <methodsynopsis role="GearmanWorker">
+   <modifier>public</modifier> <type>true</type><methodname>GearmanWorker::setOptions</methodname>
```
Comment (Girgias) [inline reference/gearman/gearmanworker/setoptions.xml:12]: Ditto

File: reference/gearman/gearmanworker/settimeout.xml [hunk 13]
```diff
@@ -9,8 +8,8 @@
 
  <refsect1 role="description">
   &reftitle.description;
-  <methodsynopsis>
-   <modifier>public</modifier> <type>bool</type><methodname>GearmanWorker::setTimeout</methodname>
+  <methodsynopsis role="GearmanWorker">
+   <modifier>public</modifier> <type>true</type><methodname>GearmanWorker::setTimeout</methodname>
```
Comment (Girgias) [inline reference/gearman/gearmanworker/settimeout.xml:12]: Ditto

**Review by Girgias — COMMENTED** [review]
This will also require regenerating the class pages to actually list all of the methods properly now

**Comment by cracksalad** [comment]
> This will also require regenerating the class pages to actually list all of the methods properly now

How to do this? Or is it just a note for the one who will merge and publish this?

Fixed all the mentioned issues now. Thanks for your advises!

**Comment by Girgias** [comment]
> > This will also require regenerating the class pages to actually list all of the methods properly now
> 
> How to do this? Or is it just a note for the one who will merge and publish this?
> 
> Fixed all the mentioned issues now. @Girgias Thanks for your advises!

Using the ``gen_stub`` script again :)

```
../php-src/build/gen_stub.php --replace-classsynopses ../path/to/pecl-networking-gearman/ reference/gearman/
```

**Comment by cracksalad** [comment]
Ok, there you go!

At first I ran into 
```
PHP Fatal error:  Uncaught Exception: Missing parent class Exception in php-src/build/gen_stub.php:3141
Stack trace:
#0 php-src/build/gen_stub.php(2835): ClassInfo->isException()
#1 php-src/build/gen_stub.php(4761): ClassInfo->getClassSynopsisElement()
#2 php-src/build/gen_stub.php(5342): replaceClassSynopses()
#3 {main}
  thrown in php-src/build/gen_stub.php on line 3141
```
caused by the `GearmanException` extending `Exception`. So I commented `GearmanException` in the stub file and updated the class page for `GearmanException` manually.

File: reference/gearman/gearmanclient.xml [hunk 14]
```diff
@@ -23,19 +22,21 @@
...
-    <xi:include xpointer="xmlns(db=http://docbook.org/ns/docbook) xpointer(id('class.gearmanclient')/db:refentry/db:refsect1[@role='description']/descendant::db:methodsynopsis[1])" />
+    <xi:include xpointer="xmlns(db=http://docbook.org/ns/docbook) xpointer(id('class.gearmanclient')/db:refentry/db:refsect1[@role='description']/descendant::db:constructorsynopsis[@role='GearmanClient'])">
+     <xi:fallback/>
+    </xi:include>
+    <xi:include xpointer="xmlns(db=http://docbook.org/ns/docbook) xpointer(id('class.gearmanclient')/db:refentry/db:refsect1[@role='description']/descendant::db:methodsynopsis[@role='GearmanClient'])">
+     <xi:fallback/>
+    </xi:include>
+    <xi:include xpointer="xmlns(db=http://docbook.org/ns/docbook) xpointer(id('class.gearmanclient')/db:refentry/db:refsect1[@role='description']/descendant::db:destructorsynopsis[@role='GearmanClient'])">
+     <xi:fallback/>
+    </xi:include>
```
[the remark this answers is not in the document]
Comment (cracksalad) [reply reference/gearman/gearmanclient.xml:39]: Does the gen_stub.php not generate documentation for destructors in general? Cause the stub file includes `__destruct()`s for those classes. 
Anyway, I will remove it and see what happens with the checks.

**Comment by Girgias** [comment]
> Ok, there you go!
> 
> At first I ran into
> 
> ```
> PHP Fatal error:  Uncaught Exception: Missing parent class Exception in php-src/build/gen_stub.php:3141
> Stack trace:
> #0 php-src/build/gen_stub.php(2835): ClassInfo->isException()
> #1 php-src/build/gen_stub.php(4761): ClassInfo->getClassSynopsisElement()
> #2 php-src/build/gen_stub.php(5342): replaceClassSynopses()
> #3 {main}
>   thrown in php-src/build/gen_stub.php on line 3141
> ```
> 
> caused by the `GearmanException` extending `Exception`. So I commented `GearmanException` in the stub file and updated the class page for `GearmanException` manually.

Right, yeah I think that requires also pointing the stub file to use the php-src stubs :/

**Comment by Girgias** [comment]
Thank you!

Would be great to document the missing methods at one point but at least the existing ones are accurate.

**Comment by cracksalad** [comment]
Thanks for your patience!

**Pull request merged**